Gemfile

```diff
@@ -12,7 +12,12 @@ end
 
 gemspec
 
-gem 'rake'
+if RUBY_VERSION < '1.9.3'
```

Comment:
Yes, dropping ruby 1.8 requires a new major version, and at least one minor version that announces the support will be deprecated.

As to whether we should... I'm not certain that using 1.8.7 will allow all the modern tooling to work anyway. I'd announce deprecation now and move to drop it shortly. Puppet 3 will be EOS in December, after all.